evm/p2p/state.py

```diff
@@ -0,0 +1,187 @@
+import asyncio
+import logging
+import random
+import time
+from typing import (  # noqa: F401
+    Any,
+    cast,
+    Dict,
+    List,
+)
+
+import rlp
+
+from trie.sync import HexaryTrieSync
+from trie.exceptions import SyncRequestAlreadyProcessed
+
+from eth_keys import datatypes  # noqa: F401
+from eth_utils import (decode_hex, encode_hex)
+
+from evm.constants import (
+    BLANK_ROOT_HASH,
+    EMPTY_SHA3,
+)
+from evm.rlp.accounts import Account
+from evm.utils.keccak import keccak
+from evm.p2p import eth
+from evm.p2p import protocol
+from evm.p2p.peer import BasePeer, ETHPeer, PeerPool
+from .constants import MAX_STATE_FETCH
+
+
+class StateDownloader:
+    logger = logging.getLogger("evm.p2p.state.StateDownloader")
+    _pending_nodes = {}  # type: Dict[Any, float]
+    _total_processed_nodes = 0
+    _report_interval = 10  # Number of seconds between progress reports.
+    # TODO: Experiment with different timeout/max_pending values to find the combination that
+    # yields the best results.
+    _max_pending = 5 * MAX_STATE_FETCH
+    _reply_timeout = 5  # seconds
+    _last_report_time = time.mktime(time.gmtime(0))
```

Comment:
interesting.  Had to look this up as my longer way to do this was always:

```python
datetime.datetime.utcfromtimestamp(0).timestamp()
```

Yours is much more concise.  That said.  Can we just initialize this at zero and call it a readability/complexity win?  Glancing around, I think that it'll be have exactly the same.